storm-server/src/main/java/org/apache/storm/daemon/supervisor/Container.java

```diff
@@ -528,28 +577,108 @@ public void cleanUpForRestart() throws IOException {
         deleteSavedWorkerUser();
         _workerId = null;
     }
-    
+
+    /**
+     * Check if the container is over its memory limit AND needs to be killed. This does not necessarily mean
+     * that it just went over the limit.
+     * @throws IOException on any error
+     */
+    public boolean isMemoryLimitViolated(LocalAssignment withUpdatedLimits) throws IOException {
```

Comment:
Yes but it is overridden by other Container Implementations and they may want to throw an exception.